integrations/oci/tls-certificates/src/main/java/io/helidon/integrations/oci/tls/certificates/DefaultOciPrivateKeyDownloader.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.tls.certificates;
+
+import java.net.URI;
+import java.security.Key;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.spec.MGF1ParameterSpec;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Objects;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.OAEPParameterSpec;
+import javax.crypto.spec.PSource;
+import javax.crypto.spec.SecretKeySpec;
+
+import io.helidon.integrations.oci.sdk.runtime.OciExtension;
+import io.helidon.integrations.oci.tls.certificates.spi.OciPrivateKeyDownloader;
+
+import com.oracle.bmc.keymanagement.KmsCryptoClient;
+import com.oracle.bmc.keymanagement.model.ExportKeyDetails;
+import com.oracle.bmc.keymanagement.requests.ExportKeyRequest;
+import com.oracle.bmc.keymanagement.responses.ExportKeyResponse;
+import jakarta.inject.Singleton;
+
+/**
+ * Implementation of the {@link OciPrivateKeyDownloader} that will use OCI's KMS to export a key.
+ */
+@Singleton
+class DefaultOciPrivateKeyDownloader implements OciPrivateKeyDownloader {
+    private final PrivateKey wrappingPrivateKey;
+    private final String wrappingPublicKeyPem;
+
+    DefaultOciPrivateKeyDownloader() {
+        try {
+            KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA");
+            generator.initialize(2048);
+            KeyPair wrappingKeyPair = generator.generateKeyPair();
+            this.wrappingPrivateKey = wrappingKeyPair.getPrivate();
+            PublicKey wrappingPublicKey = wrappingKeyPair.getPublic();
+            String pubBase64 = Base64.getEncoder().encodeToString(wrappingPublicKey.getEncoded());
+            this.wrappingPublicKeyPem = "-----BEGIN PUBLIC KEY-----" + pubBase64 + "-----END PUBLIC KEY-----";
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
wrong exception used, also why is this code happening?

Reply:
I took this code from @danielkec - Daniel, please comment.

Reply:
Temporary RSA wrapping key is being generated for wrapping temporary AES key as required by [rfc3394](https://datatracker.ietf.org/doc/html/rfc3394) used by OCI for wrapping actual key exported from OCI KMS. OCI API requires that for exporting keys from KMS, see https://docs.oracle.com/en-us/iaas/api/#/en/key/release/datatypes/ExportKeyDetails.

Reply:
So no one can steal exported key along the way, OCI returns wrapped/RSA encrypted temporary AES key concatenated together with actual (by tmp AES key encrypted) key data. Its being chopped and decrypted somewhere later in the code.

Reply:
Filed https://github.com/helidon-io/helidon/issues/7507 as a follow-up issue.